packages/components/src/components/Pagination/usePagination.tsx

```diff
@@ -0,0 +1,64 @@
+import { useMemo } from "react";
+
+const DOTS = "...";
+
+const range = (start: number, end: number): string[] => {
+  const length = end - start + 1;
+  return Array.from({ length }, (_, idx) => `${idx + start}`);
+};
+
+export const usePagination = (
```

Comment:
I think it would also be good to add unit tests for this hook that tests the different scenarios here explicitly since there's a lot going on.